nion/swift/ExportDialog.py

```diff
@@ -246,66 +250,176 @@ def cancel(self) -> bool:
+        if display_item.display_data_shape and len(display_item.display_data_shape) == 2:
+            return Geometry.IntSize(height=display_item.display_data_shape[0], width=display_item.display_data_shape[1])
+        return Geometry.IntSize(height=3, width=5)
+
+    @property
+    def width(self) -> float:
+        return self.__convert_from_pixels(self.__width)
+
+    @width.setter
+    def width(self, new_width: float) -> None:
+        self.__width = self.__convert_to_pixels(new_width)
+        self.__height = int(self.__width / self.__aspect_ratio)
+        self.notify_property_changed("width")
+        self.notify_property_changed("height")
+
+    @property
+    def height(self) -> float:
+        return self.__convert_from_pixels(self.__height)
+
+    @height.setter
+    def height(self, new_height: float) -> None:
+        self.__height = self.__convert_to_pixels(new_height)
+        self.__width = int(self.__height * self.__aspect_ratio)
+        self.notify_property_changed("width")
+        self.notify_property_changed("height")
+
+    @property
+    def units(self) -> int:
+        return self.__units.value
+
+    @units.setter
+    def units(self, new_units: int) -> None:
+        new_enum = UnitType(new_units)
+        if self.__units != new_enum:
+            self.__units = new_enum
+            self.notify_property_changed("width")
+            self.notify_property_changed("height")
+
+    def __convert_to_pixels(self, value: float) -> int:
+        return int(round(value * ConversionUnits[self.__units]))
+
+    def __convert_from_pixels(self, value: int) -> float:
+        return value / ConversionUnits[self.__units]
+
+
+class ExportSVGHandler(Declarative.Handler):
+    width_value_line_edit: UserInterface.LineEditWidget
+    height_value_line_edit: UserInterface.LineEditWidget
+
+    def __init__(self, model: ExportSizeModel) -> None:
+        super().__init__()
+        self.model = model  # Ensure model is an attribute of the handler
         u = Declarative.DeclarativeUI()
-        width_row = u.create_row(u.create_label(text=_("Width (in)"), width=80), u.create_line_edit(text="@binding(width_model.value, converter=int_converter)"), spacing=12)
-        height_row = u.create_row(u.create_label(text=_("Height (in)"), width=80), u.create_line_edit(text="@binding(height_model.value, converter=int_converter)"), spacing=12)
-        main_page = u.create_column(width_row, height_row, spacing=12, margin=12)
+        self._float_to_string_converter = Converter.FloatToStringConverter()
+
+        self.ui_view = u.create_column(
+            u.create_row(
+                u.create_label(text=_("Width:"), width=80),
+                u.create_line_edit(
+                    text="@binding(model.width, converter=_float_to_string_converter)",
+                    on_text_edited="value_updated",
+                    background_color="white",
```

Comment:
Comments: 
- Let's not change the background color. That's non-standard UI.
- If we were to change the background color, we could bind it to a value in the model.
- Instead, the model could have two values for each dimension: `width_text` and `width_placeholder_text`. Those would be updated automatically (i.e. trigger notifications) when the user set either the width or height. You might also need to include which value the user has last modified in the model.
- Then you can bind the `text` to `@binding(model.width_text)` and the `placeholder_text` to `@binding(model.placeholder_text)` and not use a converter here (the converter would be used in the model, though).